invenio/modules/relationships/engines/sqlalchemy.py

```diff
@@ -0,0 +1,117 @@
+from uuid import uuid4
+
+from sqlalchemy_utils import UUIDType
+
+from invenio.ext.sqlalchemy import db, utils
+
+
+class SQLAlchemyNode(object):
```

Comment:
I'd like to propose an alternative design: instead of relying on abstract classes, rely on the tests. The `invenio/modules/relationships/testsuite/test_nodes.py` should test that the `Node` class has all the methods a `Node` should have (the test is currently testing for the presence of `_edges`, which is wrong, since it is an implementation detail).

Any new implementation will just have to run this test to know that it's doing the right thing. Perhaps the test should be renamed to `test_node_interface.py` to stress this.